### DISCUSSION TITLE
Laser cutter not connecting

### DISCUSSION THREAD

**Discussion by eskurt** [arkypita/LaserGRBL#952] [Q&A]
When I try to press the green play button to start cutting, it's not green because the device is disconnected. It is plugged in correctly and everything has been installed. I try to connect it and it tries for a sec and then still says disconnected. How can I fix this? Thanks!

**Comment by StuartB4** [comment]
What machine do you have? Did you have to install a USB driver.
If your main controller board has a Nano board plugged in to it you
might need to install the CH340 driver. You can do that in LaserGRBL, on the tools menu.
Also check in Windows device manager to see if the Com port for the machine is there. 

If it says it's connected you can also try changing the Threading mode in LaserGRBL Settings for a slower one.

![hjhjj](https://user-images.githubusercontent.com/16980026/79058346-c4e2e780-7c64-11ea-9acf-e2aa5ad64bb0.jpg)

![Untiopopopoptled](https://user-images.githubusercontent.com/16980026/79058350-cc09f580-7c64-11ea-86d3-9a2eb48c1e0e.jpg)

**Comment by eskurt** [comment]
Hi, the laser cutter is CNC2-3040 and I’m using it on my laptop which has windows 10. I already downloaded and installed the driver, and checked the device manager. It was there, but when I go to LaserGRBL it still says disconnected. I changed it to the correct baud rate and tried, but still no luck. Will changing the threading mode affect whether I can connect? Thanks!

**Comment by eskurt** [comment]
I tried changing the threading to all the possible slower options and it still isn’t commecting

**Comment by arkypita** [comment]
Are you sure that CNC2-3040 controller use grbl as firmware?
LaserGRBL can connect only with grbl machines.

http://lasergrbl.com/faq/

**Comment by eskurt** [comment]
The laser cutter came with a file to download LaserGRBL so I assumed so, but I’ll try what the website says. If you can come up with anything else, please let me know, thank you!!

**Comment by arkypita** [comment]
> The laser cutter came with a file to download LaserGRBL so I assumed so

Ok, let's assume that it use grbl.

> I tried changing the threading to all the possible slower options and it still isn’t commecting

Thread speed doesn't have much to do with the connection problem. It may have some effect, but it actually affects most of the problems you might have with low quality arduino clones on communication stability, after connection.

It could be that your serial port needs to be "reset" on connection. Try to configure "hard reset" on the connection.

![image](https://user-images.githubusercontent.com/8782035/79107246-59883b00-7d74-11ea-80fc-a5450345bda4.png)

**Comment by arkypita** [comment]
If it does not work read this.

Unlike other programs, LaserGRBL waits for grbl (the software that runs on the machine) to send it a correct status message, before considering itself validly connected.

The message grbl should send is something like this:
<Idle|MPos:0.000,0.000,0.000|FS:0.0,0> for grbl v1.1

If LaserGRBL does not receive this message in 10 seconds, it consider itself not connected (or connected to a non grbl device).

You can log the serial session and upload here, so I can see what is the problem.

1) Run LaserGRBL
2) Click on ? menu, select "activate extended log"
3) Select a filename where to record extended log
4) Click on connect button

This will log the communication during connection phase. The format of a valid log look like this:

```
00000	00352092	log	Recording session started @ 13/04/2020 10:54:11
00001	00354790	com	Open COM5 @ 115200 baud Ctrl-X
00002	00354800	tx	[18]
00003	00354801	tx	[3F]
00004	00354829	rx	\r
00005	00354830	rx	Grbl 1.1h ['$' for help]\r
00006	00354838	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S|WCO:0.000,0.000,0.000>\r
00007	00355003	tx	[3F]
00008	00355010	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S|Ov:100,100,100>\r
00009	00355203	tx	[3F]
00010	00355209	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00011	00355402	tx	[3F]
00012	00355408	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00013	00355604	tx	[3F]
00014	00355609	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
00015	00355802	tx	[3F]
00016	00355807	rx	<Idle|MPos:0.000,0.000,0.000|FS:0,0|Pn:S>\r
[...]
```

**Comment by eskurt** [comment]
Hi,

I did that and it still won’t connect to my computer

**Comment by StuartB4** [comment]
Have a look here, there may be something to help.
https://github.com/teachables/Laserfiles/tree/master/CNC2-3040

**Comment by gmmanonymus111** [comment]
You can also try to connect to other software, like [Universal GCode Sender](https://winder.github.io/ugs_website/) to verify if your engraver is equipped with GRBL.

**Comment by eskurt** [comment]
Hi,

I tried all of the suggestions but nothing works, so I ordered an offline controller. When I press the buttons to position or move the laser, it doesn’t move. The laser doesn’t turn on either. I double checked every connection and still nothing.

**Comment by eskurt** [comment]
The laser is now on, but still now moving.

**Comment by StuartB4** [comment]
Are you using the offline controller with the USB still connected. If so try unplugging the USB first.

**Comment by eskurt** [comment]
No I had it disconnected

**Comment by gmmanonymus111** [comment]
Just to make sure you have GRBL installed properly on your arduino:
Connect your boad to your pc.
Please download [Putty](https://www.putty.org/).
Once downloaded, run it.
In the following window, set connection type to serial, then type in your arduino board properties. (Port/serial line, baud rate/speed)
![image](https://user-images.githubusercontent.com/9658243/81169109-8f49c980-8f98-11ea-94d8-de40911b6a36.png)
You can find the com port the same way StuartB4 did it in the first reply. Baud rate should be `115200 `or `9600`
Click `Open`
You should see a similar text once connected successfully: 
![image](https://user-images.githubusercontent.com/9658243/81169456-2dd62a80-8f99-11ea-9004-0c809d482f2c.png)
This means GRBL is properly installed, and you can communicate with it. By sending `?`, you should get a response like this: `<Alarm|WPos:0.000,0.000,0.000|Bf:15,128|FS:0,0|Pn:HS>`
If this is the case, LaserGRBL is at fault.

After connecting, you get something like this:
![image](https://user-images.githubusercontent.com/9658243/81170056-42ff8900-8f9a-11ea-91cc-336f0bfd31c1.png)
It means the baud rate is not correct.

In case you can not connect, you most likely have a wrong driver/USB cable.

**Comment by Cralion29** [comment]
Were you able to get a solution? I am having exactly the same problem.

**Reply by Steow** [reply]
big play button does not work on start up of the engraving process. You have to click on the little play button on the left (in front of progession"to lauch the program. After you can pause and relaunch with the big buttons. Bug or feature ?

**Reply by Tribo-2dot0** [reply]
the big Play button at the bottom left is just to unpause the pause...
